src/helpers/WLClasses.cpp

```diff
@@ -173,6 +176,9 @@ void SLayerSurface::startAnimation(bool in, bool instant) {
         alpha = in ? 1.f : 0.f;
     }
 
+    if (dimAround)
+        alpha.setUpdateCallback([this](void*) { g_pHyprRenderer->damageMonitor(g_pCompositor->getMonitorFromID(monitorID)); });
```

Comment:
the callback is not removed later. I'd put this in the layer constructor instead and in the callback itself check pLayer->dimAround

Reply:
Alright, done (unless you wanted me to get pLayer from the animatedvariable?)